src/libmongoc/tests/test-mongoc-oidc.c

```diff
@@ -797,12 +819,86 @@ PROSE_TEST(4, 5, "Reauthentication Succeeds when a Session is involved")
    test_fixture_destroy(tf);
 }
 
+PROSE_TEST(5, 1, "Azure With No Username")
+{
+   // Create URI:
+   mongoc_uri_t *uri = mongoc_uri_new("mongodb://localhost:27017/?retryReads=false");
+   {
+      mongoc_uri_set_auth_mechanism(uri, "MONGODB-OIDC");
+      bson_t props = BSON_INITIALIZER;
+      BSON_APPEND_UTF8(&props, "ENVIRONMENT", "azure");
+      char *token_resource = test_framework_getenv_required("MONGOC_AZURE_RESOURCE");
+      BSON_APPEND_UTF8(&props, "TOKEN_RESOURCE", token_resource);
+      bson_free(token_resource);
+      mongoc_uri_set_mechanism_properties(uri, &props);
+      bson_destroy(&props);
+   }
+
+   bson_error_t error;
+   mongoc_client_t *client = mongoc_client_new_from_uri_with_error(uri, &error);
+   ASSERT_OR_PRINT(client, error);
+
+   // Expect auth to succeed:
+   ASSERT_OR_PRINT(do_find(client, &error), error);
+
+   mongoc_client_destroy(client);
+   mongoc_uri_destroy(uri);
+}
+
+PROSE_TEST(5, 2, "Azure With Bad Username")
+{
+   // Create URI:
+   mongoc_uri_t *uri = mongoc_uri_new("mongodb://bad@localhost:27017/?retryReads=false");
+   {
+      mongoc_uri_set_auth_mechanism(uri, "MONGODB-OIDC");
+      bson_t props = BSON_INITIALIZER;
+      BSON_APPEND_UTF8(&props, "ENVIRONMENT", "azure");
+      char *token_resource = test_framework_getenv_required("MONGOC_AZURE_RESOURCE");
+      BSON_APPEND_UTF8(&props, "TOKEN_RESOURCE", token_resource);
+      bson_free(token_resource);
+      mongoc_uri_set_mechanism_properties(uri, &props);
+      bson_destroy(&props);
+   }
+
+   bson_error_t error;
+   mongoc_client_t *client = mongoc_client_new_from_uri_with_error(uri, &error);
+   ASSERT_OR_PRINT(client, error);
+
+   // Expect auth to fail:
+   ASSERT(!do_find(client, &error));
+   ASSERT_ERROR_CONTAINS(error, MONGOC_ERROR_CLIENT, MONGOC_ERROR_CLIENT_AUTHENTICATE, "failed");
```

Comment:
```suggestion
```
The contents of the error message is not specified by the prose test specification, only that the find command fails. Consider omitting.